Failure to build tools/benchmark/benchmark_codec_avif.cc under Ubuntu 22.04 LTS

**Describe the bug**

Libjxl and submodules are updated to the latest as of August 15, 2024.  The tools/benchmark/benchmark_codec_avif.cc source file fails to compile due to some incompatibility with libavif.  Libjxl was previously compiled successfully on July 14, 2024.

**To Reproduce**

The build is done in a 'build' subdirectory using

```
cmake -DCMAKE_INSTALL_RPATH=/usr/local/lib -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF ..
```

**Expected behavior**

I expect the software to compile (as it used to do).

**Screenshots**

```
[ 97%] Building CXX object tools/CMakeFiles/benchmark_xl.dir/benchmark/benchmark_codec_avif.cc.o
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc: In function ‘jxl::Status jpegxl::tools::{anonymous}::SetUpAvifColor(const jxl::ColorEncoding&, bool, avifImage*)’:
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:129:31: error: void value not ignored as it ought to be
  129 |         avifImageSetProfileICC(image, color.ICC().data(), color.ICC().size()));
      |                               ^
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:23:51: note: in definition of macro ‘JXL_RETURN_IF_AVIF_ERROR’
   23 |     avifResult jxl_return_if_avif_error_result = (result);                     \
      |                                                   ^~~~~~
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc: In member function ‘jxl::Status jpegxl::tools::AvifCodec::Compress(const string&, const jxl::CodecInOut*, jxl::ThreadPool*, std::vector<unsigned char>*, jpegxl::tools::SpeedStats*)’:
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:313:60: error: void value not ignored as it ought to be
  313 |         JXL_RETURN_IF_AVIF_ERROR(avifRGBImageAllocatePixels(&rgb_image));
      |                                                            ^
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:23:51: note: in definition of macro ‘JXL_RETURN_IF_AVIF_ERROR’
   23 |     avifResult jxl_return_if_avif_error_result = (result);                     \
      |                                                   ^~~~~~
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc: In member function ‘jxl::Status jpegxl::tools::AvifCodec::Decompress(const string&, jxl::Span<const unsigned char>, jxl::ThreadPool*, jxl::CodecInOut*, jpegxl::tools::SpeedStats*)’:
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:382:60: error: void value not ignored as it ought to be
  382 |         JXL_RETURN_IF_AVIF_ERROR(avifRGBImageAllocatePixels(&rgb_image));
      |                                                            ^
/home/bfriesen/src/libjxl/tools/benchmark/benchmark_codec_avif.cc:23:51: note: in definition of macro ‘JXL_RETURN_IF_AVIF_ERROR’
   23 |     avifResult jxl_return_if_avif_error_result = (result);                     \
      |                                                   ^~~~~~
```

**Environment**
 - OS: Ubuntu 22.04 LTS
 - Compiler version: GCC 11.4.0
 - CPU type: x86_64
 - cjxl/djxl version string: Unknown

**Additional context**

Installed Libavif version is 0.9.3-3

<!--
Currently github does not allow uploading files that end in `.jxl`, but when you
rename them for example as `image.jxl.jpg`, it will be possible to upload them
and also view them in browsers that are configured to support it.

See https://github.com/orgs/github-community/discussions/18139
-->

--- a/tools/benchmark/benchmark_codec_avif.cc
+++ b/tools/benchmark/benchmark_codec_avif.cc
@@ -125,8 +125,12 @@ Status SetUpAvifColor(const ColorEncoding& color, bool rgb,
   }
 
   if (need_icc) {
+#if AVIF_VERSION_MAJOR < 1
+    avifImageSetProfileICC(image, color.ICC().data(), color.ICC().size());
+#else
     JXL_RETURN_IF_AVIF_ERROR(
         avifImageSetProfileICC(image, color.ICC().data(), color.ICC().size()));
+#endif
   }
   return true;
 }
@@ -310,7 +314,11 @@ class AvifCodec : public ImageCodec {
         avifRGBImageSetDefaults(&rgb_image, image.get());
         rgb_image.format =
             ib.HasAlpha() ? AVIF_RGB_FORMAT_RGBA : AVIF_RGB_FORMAT_RGB;
+#if AVIF_VERSION_MAJOR < 1
+        avifRGBImageAllocatePixels(&rgb_image);
+#else
         JXL_RETURN_IF_AVIF_ERROR(avifRGBImageAllocatePixels(&rgb_image));
+#endif
         std::unique_ptr<avifRGBImage, void (*)(avifRGBImage*)> pixels_freer(
             &rgb_image, &avifRGBImageFreePixels);
         const double start_convert_image = jxl::Now();
@@ -379,7 +387,11 @@ class AvifCodec : public ImageCodec {
         avifRGBImageSetDefaults(&rgb_image, decoder->image);
         rgb_image.format =
             has_alpha ? AVIF_RGB_FORMAT_RGBA : AVIF_RGB_FORMAT_RGB;
+#if AVIF_VERSION_MAJOR < 1
+        avifRGBImageAllocatePixels(&rgb_image);
+#else
         JXL_RETURN_IF_AVIF_ERROR(avifRGBImageAllocatePixels(&rgb_image));
+#endif
         std::unique_ptr<avifRGBImage, void (*)(avifRGBImage*)> pixels_freer(
             &rgb_image, &avifRGBImageFreePixels);
         JXL_RETURN_IF_AVIF_ERROR(avifImageYUVToRGB(decoder->image, &rgb_image));

